source/03-components/table/table--sortable.es6.ts

```diff
@@ -0,0 +1,295 @@
+    }
+
+    const sortAnnouncement = `The table${tableName} is now sorted by ${headerLabel} in ${
+      sortedAscending ? ASCENDING : DESCENDING
+    } order.`;
+    liveRegion.innerText = sortAnnouncement;
+  } else {
+    // eslint-disable-next-line no-console
+    console.error(
+      'Table containing a sortable column header is not followed by an aria-live region.'
+    );
+  }
+};
+
+/**
+ * Toggle a header’s sort state, optionally providing a target state. If no
+ * state is provided, the current state will be toggled (from false to true,
+ * and vice-versa).
+ *
+ * @param {HTMLTableCellElement} header
+ * @param {boolean?} isAscending
+ */
+const toggleSort = (
+  header: HTMLTableCellElement,
+  isAscending: boolean
+): void => {
+  const table = header.closest(TABLE) as HTMLTableElement;
+  let safeAscending = isAscending;
+
+  if (typeof safeAscending !== 'boolean') {
+    safeAscending = header.getAttribute(SORTED) === ASCENDING;
+  }
+
+  if (!table) {
+    throw new Error(`${SORTABLE_HEADER} is missing outer ${TABLE}`);
+  }
+
+  safeAscending = sortRows(header, isAscending);
+
+  if (safeAscending) {
+    getColumnHeaders(table).forEach((otherHeader: HTMLTableCellElement) => {
+      if (otherHeader !== header) {
+        unsetSort(otherHeader);
+      }
+    });
+    updateLiveRegion(table, header);
+  }
+};
+
+/**
+ ** Inserts a button with icon inside a sortable header.
+ * @param {HTMLTableCellElement} header
+ * @param {string} imagePath
+ */
+
+const createHeaderButton = (
+  header: HTMLTableCellElement,
+  imagePath: string
+): void => {
+  const buttonEl = document.createElement('button');
+  buttonEl.setAttribute('tabindex', '0');
+  buttonEl.classList.add(SORT_BUTTON_CLASS);
+
+  // Add button icons.
+  buttonEl.innerHTML = `
+        <svg class="c-icon c-table__sort-icon"><use xlink:href="${imagePath}/sprite.artifact.svg#sort"></use></svg>
+        <svg class="c-icon c-table__sorted-icon"><use xlink:href="${imagePath}/sprite.artifact.svg#sorted"></use></svg>
+      `;
+  const headerInnerElem = document.createElement('div');
+  headerInnerElem.innerHTML = header.innerHTML;
+  headerInnerElem.classList.add('c-table__header-inner');
+  headerInnerElem.appendChild(buttonEl);
+  header.innerHTML = '';
+  header.appendChild(headerInnerElem);
+  updateSortLabel(header);
+};
+
+Drupal.behaviors.sortableTable = {
+  attach(context, settings) {
+    const { imagePath } = settings.gesso;
+
+    const tables = once(
+      'sort-buttons',
+      '.c-table.is-sortable',
+      context
+    ) as NodeListOf<HTMLElement>;
+
+    tables.forEach(table => {
+      // Add sortable attribute to all column headers if wanted.
+      if (SORT_ALL_COLUMNS) {
+        table.querySelectorAll<HTMLTableCellElement>('thead th').forEach(th => {
+          th.setAttribute(SORTABLE_ATTRIBUTE, '');
+        });
+      }
+
+      // Add aria-live region for sort announcements.
+      const liveRegion = table.nextElementSibling as HTMLElement;
```

Comment:
Removing casting as we do a check another on the next line to see if it is `null`.